### ISSUE TITLE
azuredevops_git_repository_file does not consider branch while checking for file existence

### ISSUE THREAD

**Issue by bloewes** [microsoft/terraform-provider-azuredevops#663]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and Azure DevOps Provider) Version

Terraform 1.3.2
Azure Devops Provider 0.3.0

### Affected Resource(s)

resource azuredevops_git_repository_file

### Error description

I have repository with a default branch main. If I create resource as follows

```
   resource "azuredevops_git_repository_file" "template" {
  repository_id      = data.azuredevops_git_repository.project_repo.id
  file                      = "test.txt"
  content               = "foo"
  branch                 = "refs/heads/terraform"
  commit_message = "Update"
  overwrite_on_create = true
```
I expect the file to be created in my branch terraform. Note that this branch is empty so far.
However, my default branch main already contains a file "test.txt".
If I look at the following code line https://github.com/microsoft/terraform-provider-azuredevops/blob/270599b9a42e85380a0ac0e48ab164be922238fc/azuredevops/internal/service/git/resource_git_repository_file.go#L109-L112
and at the corresponding function GetItem, it doesn't take the branch into account, it always checks the default branch where the file already exists.

Therefore, changeType gets set to Edit not Add and the apply of the resource fails.
https://github.com/microsoft/terraform-provider-azuredevops/blob/270599b9a42e85380a0ac0e48ab164be922238fc/azuredevops/internal/service/git/resource_git_repository_file.go#L118-L124

Can you fix this call to take the branch into account correctly?

**Comment by bloewes** [comment]
@xuzhang3 any chance you can have a look at this?

**Comment by xuzhang3** [comment]
@bloewes  The create/update will failed if the branch not exist.  Introduce a new resource `azuredevops_repositroy_branche` to managed the branches would be the best choice.

**Comment by bloewes** [comment]
Fixed by #713
